### PR TITLE
Add new ubuntu based dockerimage and fix docker-entrypoint_centos.sh

### PR DESCRIPTION
**Pull request by foxzi**
Added new docker file for building ubuntu image

### CODE REVIEW

File: dockerfiles/ubuntu_20.04 [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    libproj-dev \
+    gosu \
+    curl \
+    ca-certificates \
+    && ln -snf /usr/share/zoneinfo/UTC /etc/localtime && echo UTC > /etc/timezone \
+    && ln -s /usr/lib/x86_64-linux-gnu/libprofiler.so.0.4.18 /usr/lib/x86_64-linux-gnu/libprofiler.so \
+    && curl -L https://tarantool.io/pre-release/2/installer.sh | bash \
+    && apt -y install tarantool \
+    && apt clean \
+    && set -x && tarantoolctl rocks install lua-term \
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:50]: nit: add `newline` after set-x command

File: dockerfiles/ubuntu_20.04 [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    libssl-dev \
+    libpq-dev \
+    gis-devel \
+    libproj-dev \
+    gosu \
+    curl \
+    ca-certificates \
+    && ln -snf /usr/share/zoneinfo/UTC /etc/localtime && echo UTC > /etc/timezone \
+    && ln -s /usr/lib/x86_64-linux-gnu/libprofiler.so.0.4.18 /usr/lib/x86_64-linux-gnu/libprofiler.so \
+    && curl -L https://tarantool.io/pre-release/2/installer.sh | bash \
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:47]: `pre-release`. probably it should be `release` at the moment

File: dockerfiles/ubuntu_20.04 [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    LUAROCK_TARANTOOL_GPERFTOOLS_VERSION=1.0.1
+
+ARG ENABLE_BUNDLED_LIBYAML \
+    LUAJIT_DISABLE_SYSPROF \
+    LUAJIT_ENABLE_GC64
+
+RUN set -x && apt update && apt -y --no-install-recommends install libgoogle-perftools4 \
+    libgoogle-perftools4 \
+    deborphan \
+    unzip \
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:35]: I would remove packages like unzip and curl with dependencies after installation to reduce image size

Comment (artembo) [reply dockerfiles/ubuntu_20.04:35]: please check other redundant packages which can be removed for the same reason

File: dockerfiles/ubuntu_20.04 [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    LUAROCK_TARANTOOL_MQTT_VERSION=1.5.1 \
+    LUAROCK_TARANTOOL_GIS_VERSION=1.0.1 \
+    LUAROCK_TARANTOOL_PROMETHEUS_VERSION=1.0.4 \
+    LUAROCK_TARANTOOL_GPERFTOOLS_VERSION=1.0.1
+
+ARG ENABLE_BUNDLED_LIBYAML \
+    LUAJIT_DISABLE_SYSPROF \
+    LUAJIT_ENABLE_GC64
+
+RUN set -x && apt update && apt -y --no-install-recommends install libgoogle-perftools4 \
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:32]: Please take a look at `--no-install-suggests` — it will help to reduce the number of installed packages

**Review by artembo — CHANGES_REQUESTED** [review]

File: files/docker-entrypoint_centos.sh [hunk 5]
```diff
@@ -3,7 +3,7 @@ set -e
 
 # first arg is `-f` or `--some-option`
 # or first arg is `something.conf`
-if [ "${1:0:1}" = '-' ]; then
+if [ " -${1#?}" = " $1" ]; then
```
Comment (artembo) [inline files/docker-entrypoint_centos.sh:6]: Could you please explain the reason of this change in the commit message and PR description

File: dockerfiles/ubuntu_20.04 [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    libgoogle-perftools4 \
+    deborphan \
+    unzip \
+    libgeos-dev \
+    libsasl2-dev \
+    libssl-dev \
+    libpq-dev \
+    gis-devel \
+    libproj-dev \
+    gosu \
```
[the remark this answers is not in the document]
Comment (foxzi) [reply dockerfiles/ubuntu_20.04:43]: In ubuntu packeges presents only gosu

File: dockerfiles/ubuntu_20.04 (hunk 2 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: fixed

File: dockerfiles/ubuntu_20.04 (hunk 4 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: If Yoo look above this option already set

File: files/docker-entrypoint_centos.sh (hunk 5 above)
Comment (foxzi) [reply files/docker-entrypoint_centos.sh:6]: This fix helps working script in ubuntu and centos.

File: dockerfiles/ubuntu_20.04 (hunk 4 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: done

File: dockerfiles/ubuntu_20.04 [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    && tarantoolctl rocks install http $LUAROCK_HTTP_VERSION \
+    && tarantoolctl rocks install pg $LUAROCK_TARANTOOL_PG_VERSION \
+    && tarantoolctl rocks install mysql $LUAROCK_TARANTOOL_MYSQL_VERSION \
+    && tarantoolctl rocks install memcached $LUAROCK_MEMCACHED_VERSION \
+    && tarantoolctl rocks install metrics $LUAROCK_METRICS_VERSION \
+    && tarantoolctl rocks install prometheus $LUAROCK_TARANTOOL_PROMETHEUS_VERSION \
+    && tarantoolctl rocks install mqtt $LUAROCK_TARANTOOL_MQTT_VERSION \
+    && tarantoolctl rocks install gis $LUAROCK_TARANTOOL_GIS_VERSION \
+    && tarantoolctl rocks install gperftools $LUAROCK_TARANTOOL_GPERFTOOLS_VERSION \
+    && apt purge -y unzip \
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:60]: what about curl?

File: dockerfiles/ubuntu_20.04 [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    LUAROCK_HTTP_VERSION=1.2.0 \
+    LUAROCK_MEMCACHED_VERSION=1.0.1 \
+    LUAROCK_METRICS_VERSION=0.12.0 \
+    LUAROCK_TARANTOOL_PG_VERSION=2.0.2 \
+    LUAROCK_TARANTOOL_MYSQL_VERSION=2.1.0 \
+    LUAROCK_TARANTOOL_MQTT_VERSION=1.5.1 \
+    LUAROCK_TARANTOOL_GIS_VERSION=1.0.1 \
+    LUAROCK_TARANTOOL_PROMETHEUS_VERSION=1.0.4 \
+    LUAROCK_TARANTOOL_GPERFTOOLS_VERSION=1.0.1 \
+    GC64=false
```
Comment (artembo) [inline dockerfiles/ubuntu_20.04:24]: It has to be overridden during the build process in case we need gc64 version, so we have to use build args here.

**Review by artembo — CHANGES_REQUESTED** [review]

File: dockerfiles/ubuntu_20.04 (hunk 7 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: fixed

File: dockerfiles/ubuntu_20.04 (hunk 8 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: Move GC64 variable to ARG

**Review by artembo — APPROVED** [review]
LGTM

**Pull request merged**